medicines/web/src/components/search-filter/index.tsx

```diff
@@ -65,16 +73,61 @@ const DocTypeCheckbox: React.FC<IDocTypeCheckboxProps> = props => {
 };
 
 const SearchFilter: React.FC<ISearchFilterProps> = props => {
+  const [checkedFilters, setCheckedFilters] = React.useState(
+    props.currentlyEnabledDocTypes,
+  );
+  const submitButton = useRef(null);
+  const filterHeader = useRef(null);
+
   const generateCheckboxFor = (docType: DocType, name: string) => (
-    <DocTypeCheckbox {...props} docTypeForThisCheckbox={docType} name={name} />
+    <DocTypeCheckbox
+      toggleDocType={toggleDocType}
+      currentlyEnabledDocTypes={checkedFilters}
+      docTypeForThisCheckbox={docType}
+      name={name}
+    />
   );
 
+  const toggleDocType = docTypeToToggle => {
+    const enabledDocTypes = Array.from(checkedFilters);
+    if (enabledDocTypes.includes(docTypeToToggle)) {
+      const docTypeIndex = enabledDocTypes.indexOf(docTypeToToggle);
+      enabledDocTypes.splice(docTypeIndex, 1);
+    } else {
+      enabledDocTypes.push(docTypeToToggle);
+    }
+    setCheckedFilters(enabledDocTypes);
+  };
+
+  const submit = e => {
+    e.preventDefault();
+    props.updateDocTypes(checkedFilters);
+  };
+
+  useEffect(() => {
+    if (
+      props.rerouteType != null &&
+      RerouteType[props.rerouteType.toString()] === RerouteType.CheckboxSelected
+    ) {
+      filterHeader.current?.scrollIntoView();
+      submitButton.current?.focus();
+    }
+  }, [props.rerouteType]);
+
   return (
     <StyledSearchFilter>
-      <h2>Filter documents by</h2>
+      <h2 ref={filterHeader}>Filter documents by</h2>
       {generateCheckboxFor(DocType.Spc, 'Summary of Product Characteristics')}
       {generateCheckboxFor(DocType.Pil, 'Patient Information Leaflet')}
       {generateCheckboxFor(DocType.Par, 'Public Assessment Reports')}
+      <ButtonContainer>
```

Comment:
I see this was an extra addition we did not see during training to add some padding, nice.